metrics/auto_increment/auto_increment.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 Block, Inc.
+
+package autoincrement
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/cashapp/blip"
+)
+
+const (
+	DOMAIN = "autoinc"
+
+	OPT_EXCLUDE = "exclude"
+	OPT_INCLUDE = "include"
+)
+
+// AutoIncrement collects auto-increment utilization for the autoinc domain.
+// https://dev.mysql.com/doc/refman/8.0/en/sys-schema-auto-increment-columns.html
+type AutoIncrement struct {
+	db *sql.DB
+	// --
+	query  map[string]string
+	params map[string][]interface{}
+}
+
+// Verify collector implements blip.Collector interface.
+var _ blip.Collector = &AutoIncrement{}
+
+// NewAutoIncrement makes a new AutoIncrement collector,
+func NewAutoIncrement(db *sql.DB) *AutoIncrement {
+	return &AutoIncrement{
+		db:     db,
+		query:  map[string]string{},
+		params: map[string][]interface{}{},
+	}
+}
+
+// Domain returns the Blip metric domain name (DOMAIN const).
+func (t *AutoIncrement) Domain() string {
+	return DOMAIN
+}
+
+// Help returns the output for blip --print-domains.
+func (t *AutoIncrement) Help() blip.CollectorHelp {
+	return blip.CollectorHelp{
+		Domain:      DOMAIN,
+		Description: "Auto Increment Utilization",
+		Options: map[string]blip.CollectorHelpOption{
+			OPT_INCLUDE: {
+				Name: OPT_INCLUDE,
+				Desc: "Comma-separated list of database or table names to include (overrides option " + OPT_EXCLUDE + ")",
+			},
+			OPT_EXCLUDE: {
+				Name:    OPT_EXCLUDE,
+				Desc:    "Comma-separated list of database or table names to exclude (ignored if " + OPT_INCLUDE + " is set)",
+				Default: "mysql.*,information_schema.*,performance_schema.*,sys.*",
+			},
+		},
+		Groups: []blip.CollectorKeyValue{
+			{Key: "db", Value: "the database name for the corresponding auto increment value"},
+			{Key: "tbl", Value: "the table name for the corresponding auto increment value"},
+			{Key: "col", Value: "the column name for the corresponding auto increment value"},
+			{Key: "data_type", Value: "the data type of the column"},
```

Comment:
Is this needed?

Reply:
We care about the distinction between `int` and `bigint` for things like triage. Whether or not it is signed also informs us about possible solutions. In terms of monitoring we may want to alert on things that are not `unsigned` or are `int` instead of `bigint`. We could split those into separate tags if we wanted.